jerry-core/ecma/operations/ecma-regexp-object.c

```diff
@@ -535,6 +543,9 @@ re_match_regexp (re_matcher_ctx_t *re_ctx_p, /**< RegExp matcher context */
           if (!sub_str_p)
           {
             match_value = re_match_regexp (re_ctx_p, bc_p, str_curr_p, &sub_str_p);
+#ifdef REGEXP_RECURSION_LIMIT
+            --re_ctx_p->recursion_depth;
```

Comment:
I thought we had much less `--re_ctx_p->recursion_depth` this way. This doesn't looks simpler unfortunately. Maybe if we would introduce a macro/inline-function for re_match_regexp which reduces recursion_depth, that could simplify things. I am not sure. What others think?